### ISSUE TITLE
bug in `plotnine 0.7.0`

### ISSUE THREAD

**Issue by jbloom** [has2k1/plotnine#403]
I get the following error:
```
    152         # 1.45 makes the default colourbar wider than the
    153         # legend entry boxes.
--> 154         width = (self.barwidth or _d('legend_key_width')) * 1.45
    155         height = (self.barheight or _d('legend_key_height')) * 1.45
    156         height *= 5

TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
```

**Comment by has2k1** [comment]
@jbloom, I cannot create the error, can you please post a minimal example.

**Comment by jbloom** [comment]
@has2k1, here is code to produce the error with `plotnine` 0.7.0 (`pandas` 1.0.5 in case that matters):

```
import pandas as pd

from plotnine import *

df = pd.DataFrame.from_records(
            [(0.0, -3.0), (-15.0, -10.0), (-6.0, -7.0),
             (-6.0, -6.0), (-15.0, -15.0)],
            columns=['predicted', 'actual']
            )

ggplot(df, aes('actual', 'predicted')) + geom_bin2d()
```

If I run the above in a Jupyter notebook, I get the following error:

```
TypeError                                 Traceback (most recent call last)
/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/IPython/core/formatters.py in __call__(self, obj)
    700                 type_pprinters=self.type_printers,
    701                 deferred_pprinters=self.deferred_printers)
--> 702             printer.pretty(obj)
    703             printer.flush()
    704             return stream.getvalue()

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    392                         if cls is not object \
    393                                 and callable(cls.__dict__.get('__repr__')):
--> 394                             return _repr_pprint(obj, self, cycle)
    395 
    396             return _default_pprint(obj, self, cycle)

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    698     """A pprint that just redirects to the normal repr function."""
    699     # Find newlines and replace them with p.break_()
--> 700     output = repr(obj)
    701     lines = output.splitlines()
    702     with p.group():

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/ggplot.py in __repr__(self)
     86         # in the jupyter notebook.
     87         if not self.figure:
---> 88             self.draw()
     89         plt.show()
     90         return '<ggplot: (%d)>' % self.__hash__()

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/ggplot.py in draw(self, return_ggplot)
    179         # new frames knowing that they are separate from the original.
    180         with pd.option_context('mode.chained_assignment', None):
--> 181             return self._draw(return_ggplot)
    182 
    183     def _draw(self, return_ggplot=False):

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/ggplot.py in _draw(self, return_ggplot)
    210             if self.figure is not None:
    211                 plt.close(self.figure)
--> 212             raise err
    213 
    214         if return_ggplot:

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/ggplot.py in _draw(self, return_ggplot)
    202                 self._draw_labels()
    203                 self._draw_breaks_and_labels()
--> 204                 self._draw_legend()
    205                 self._draw_title()
    206                 self._draw_watermarks()

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/ggplot.py in _draw_legend(self)
    421         Draw legend onto the figure
    422         """
--> 423         legend_box = self.guides.build(self)
    424         if not legend_box:
    425             return

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/guides/guides.py in build(self, plot)
    142             return
    143 
--> 144         gboxes = self.draw(gdefs, plot.theme)
    145         bigbox = self.assemble(gboxes, gdefs, plot.theme)
    146         return bigbox

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/guides/guides.py in draw(self, gdefs, theme)
    284             g.theme = theme
    285             g._set_defaults()
--> 286         return [g.draw() for g in gdefs]
    287 
    288     def assemble(self, gboxes, gdefs, theme):

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/guides/guides.py in <listcomp>(.0)
    284             g.theme = theme
    285             g._set_defaults()
--> 286         return [g.draw() for g in gdefs]
    287 
    288     def assemble(self, gboxes, gdefs, theme):

/fh/fast/bloom_j/software/conda_v2/envs/plotnine_0.7.0/lib/python3.6/site-packages/plotnine/guides/guide_colorbar.py in draw(self)
    152         # 1.45 makes the default colourbar wider than the
    153         # legend entry boxes.
--> 154         width = (self.barwidth or _d('legend_key_width')) * 1.45
    155         height = (self.barheight or _d('legend_key_height')) * 1.45
    156         height *= 5

TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
```

**Comment by has2k1** [comment]
I still cannot reproduce it. This is what I get (python 3.6, pandas 1.0.5) .

```python
import pandas as pd

from plotnine import *

df = pd.DataFrame.from_records(
            [(0.0, -3.0), (-15.0, -10.0), (-6.0, -7.0),
             (-6.0, -6.0), (-15.0, -15.0)],
            columns=['predicted', 'actual']
            )

ggplot(df, aes('actual', 'predicted')) + geom_bin2d()
```

![nobug](https://user-images.githubusercontent.com/780341/85453271-61e6c880-b5a4-11ea-8591-a9dff4cf1d63.png)

**Comment by jbloom** [comment]
@has2k1, I figured out what causes the above bug to appear. It only happens with specific `plotnine` themes. For instance, this code gives the error mentioned in my post above. Note that this code differs from the code you ran without error in previous post only by using a different theme.

So I think in `plotnine` 0.7.0, `geom_bin2d` is broken for specific themes:

```
import pandas as pd
  
from plotnine import *

df = pd.DataFrame.from_records(
            [(0.0, -3.0), (-15.0, -10.0), (-6.0, -7.0),
             (-6.0, -6.0), (-15.0, -15.0)],
            columns=['predicted', 'actual']
            )

theme_set(theme_matplotlib())

p = ggplot(df, aes('actual', 'predicted')) + geom_bin2d()

p.save('plot.pdf')
```

**Comment by jbloom** [comment]
@has2k1, do you plan to cut a new release with this bug fix (0.7.1)? Just curious as I'm using a bunch of code with matplotlib theme built on, so this bug is keeping me from having to stay pinned to version 0.6.

**Comment by has2k1** [comment]
v0.7.1 is out.
